src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -117,27 +122,82 @@ internal void ThrowNoInvokeException()
                 return null;
             }
 
-            StackAllocedArguments stackArgs = default;
-            Span<object?> arguments = default;
-            if (actualCount != 0)
+            Debug.Assert(obj != null);
+
+            unsafe
             {
-                arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, ArgumentTypes);
-            }
+                if (argCount == 0)
+                {
+                    Invoker.InvokeUnsafe(obj, args: default, invokeAttr);
+                }
+                else
+                {
+                    Debug.Assert(parameters != null);
+                    Span<object?> parametersOut;
+                    bool copyBack = Invoker.HasRefs;
 
-            object? retValue = InvokeWorker(obj, invokeAttr, arguments);
+                    if (argCount <= MaxStackAllocArgCount)
+                    {
+                        StackAllocatedByRefs byrefStorage = default;
+                        IntPtr** unsafeByrefParameters = (IntPtr**)&byrefStorage;
+                        StackAllocedArguments argStorage = default;
+                        parametersOut = new Span<object?>(ref argStorage._arg0, argCount);
 
-            // copy out. This should be made only if ByRef are present.
-            // n.b. cannot use Span<T>.CopyTo, as parameters.GetType() might not actually be typeof(object[])
-            for (int index = 0; index < arguments.Length; index++)
-            {
-                parameters![index] = arguments[index];
-            }
+                        CheckArguments(
+                            ref parametersOut,
+                            unsafeByrefParameters,
+                            ref copyBack,
+                            parameters,
+                            ArgumentTypes,
+                            binder,
+                            culture,
+                            invokeAttr);
 
-            return retValue;
+                        Invoker.InvokeUnsafe(obj, unsafeByrefParameters, invokeAttr);
+                    }
+                    else
+                    {
+                        parametersOut = new Span<object?>(new object[argCount]);
+                        IntPtr** unsafeByrefParameters = stackalloc IntPtr*[argCount];
```

Comment:
any concern that argCount might be large enough to cause stack overflow here (and other stackalloc places)?

Reply:
The argument count has been validated to match a real method by this point. The same stackalloc is done on the unmanaged side of the runtime today. This refactoring is moving around where the stackalloc is done, so it is not introducing anything new.

If you use methods with very large number of arguments, you are generally at higher risk of hitting stackoverflows.